omniport/core/open_auth/views/retrieve_data.py

```diff
@@ -0,0 +1,105 @@
+from rest_framework import status, generics
+from rest_framework.response import Response
+from oauth2_provider.models import AccessToken
+
+from kernel.models import Person
+from open_auth.utils import get_field_data, get_roles, get_display_picture
+
+
+MODEL_REGEX = [
+    'person',
+    'student',
+    'faculty_member',
+    'biological_information',
+    'contact_information'
+]
+MODEL_STRINGS = [
+    'person',
+    'person.student',
+    'person.facultymember',
+    'person.biologicalinformation',
+    'person.contact_information.first()'
+]
+
+
+class GetUserData(generics.GenericAPIView):
+    """
+    View to retrieve data for OAuth based applications
+    """
+
+    def get(self, request):
+        """
+        View to serve GET requests
+        :param request: the request that is to be responded to
+        :return: the response for request
+        """
+
+        token = request.headers['Authorization'].replace('OAUTH_TOKEN ', '')
+
+        try:
+            access_token = AccessToken.objects.get(token=token)
+            application = access_token.application
+        except AccessToken.DoesNotExist:
+            return Response(
+                data="Please provide access token in the headers",
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        if not access_token.is_valid():
+            return Response(
+                data="Token entered is invalid",
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        if access_token.is_expired():
+            return Response(
+                data="This token has expired",
+                status=status.HTTP_401_UNAUTHORIZED
+            )
+
+        app_data_points = application.data_points
+        user = access_token.user
+        response_data = {}
+
+        try:
+            person = user.person
+        except Person.DoesNotExist:
+            response_data['username'] = user.username
+            return Response(
+                data=response_data,
+                status=status.HTTP_200_OK
+            )
+
+        response_data['username'] = user.username
+
+        for model_name, model_string in zip(MODEL_REGEX, MODEL_STRINGS):
+            model_data_points = [x.split('.', 1)[1] for
```

Comment:
We follow a different indentation style. See second style in [PEP 8](https://www.python.org/dev/peps/pep-0008/#indentation)
```python
x = [
    ...
    ...
]
```